lib/peg.d.ts

```diff
@@ -10,6 +10,8 @@ declare namespace PEG {
   }
 
   interface LocationRange {
+    /** Any object that was supplied to the `parse()` call as the `source` option. */
+    source: any;
```

Comment:
not string?

Reply:
or maybe `string|URL`

Reply:
The question I guess is how it will be used in reporting in the GrammarError, to which I don't see changes yet.

Reply:
`GrammarError` contains reference to the location and this location will contain reference to the source:
```js
GrammarError = {
  message: "...",
  expected: [],
  found: "...", 
  location: {
    source: any value supplied by the user,
    start: { ... },
    end: { ... },
  }
}
```
Any value is acceptable, so you can, for example, pass the node Stream object of File object. So it will be easy to use Peggy in various VFS environments. Peggy itself will not inspect that value, it is completely opaque for it.

I tried to describe that in the comments, but maybe they not so clear as I think -- I'm not a native speaker and I think that my English is not so good. So any corrections welcome!

Reply:
What I'm thinking about is what happens when we have included grammars.  Possibly the GrammerException `message` and definitely the output of the CLI will want to include information about the source, or putting this in won't be as helpful.

Reply:
I think, that in any case `GrammarError.message` shouldn't contain a line, column or file (or another source) information. All that stuff should be added by the application that's used the parser and it already known the type of the `source` object so can extract any necessary information. In the simplest case this would be just a string with the file name, as right now I done in the `bin/peggy`.

So the parser itself will never work with the `source` and might not require any specific type for it.

Reply:
That makes sense to me.  Should we go through all of the places where we're setting line/col in message, and remove that info, then make sure `bin/peggy` is outputting it?  I'll add a bug for that, which you can either catch with this PR or we'll do a separate change, whatever you prefer.

Reply:
See #102

Reply:
I will not mix up line/col change with that, because need at least check how website uses it.